File: pyiArduinoI2Cencoder/cpp/EncTurnLeftRight.cpp
/*
 *	ПРИМЕР ЧТЕНИЯ ТАКТОВ ПОВОРОТА ЭНКОДЕРА:
 * В данном примере считываются такты поворотов
 * энкодера, отдельно вправо и отдельно влево.
 *
 */

#include <iostream>

//  Подключаем библиотеку для работы с энкодером I2C-flash.
#include "../iarduino_I2C_Encoder.h"

//  Инстанциируем объект, указывая адрес модуля на шине I2C.
iarduino_I2C_Encoder enc(0x09);

void loop(void);

int main()
{
	//  Инициируем работу с энкодером.
	enc.begin();
	for (;;)
		loop();
}

void loop()
{
	int turnL=0, turnR=0;

	/*	Считываем такты поворота энкодера:	*/

	turnL=enc.getEncoder(ENC_TURN_LEFT);
	turnR=enc.getEncoder(ENC_TURN_RIGHT);

	//  Выводим считанные данные:
	if (turnL) {
		std::cout << "-";
		std::cout << turnL;
		std::cout << '\n';
	}
	if (turnR) {
		std::cout << "+" ;
		std::cout << turnR ;
		std::cout << '\n';
	}

	delay(100);
}

File: pyiArduinoI2Cencoder/examples/EncTurnLeftRight.py
# ПРИМЕР ЧТЕНИЯ ТАКТОВ ПОВОРОТА ЭНКОДЕРА:
# В данном примере считываются такты поворотов энкодера,
# отдельно вправо и отдельно влево.

from time import sleep

# Подключаем библиотеку для работы с энкодером I2C-flash.
from pyiArduinoI2Cencoder import *
# Инстанцируем объект, указывая адрес модуля на шине I2C.
enc = pyiArduinoI2Cencoder(0x09)

print("Вращайте барабан!")

while True:

    # Определяем переменные для чтения количества тактов поворота энкодера.
    turnL, turnR = 0, 0

    # Считываем такты поворота энкодера:
    turnL=enc.getEncoder(ENC_TURN_LEFT)   # Считываем количество влево
    turnR=enc.getEncoder(ENC_TURN_RIGHT)  # Считываем количество вправо

    # Выводим считанные данные:
    if turnL:
        print("-",
             turnL)

    if turnR:
        print("+",
             turnR)

    sleep(.1)
